### PR TITLE
Fix and enable remaining navigation block e2e tests

### PR DESCRIPTION
**Pull request by talldan**
## Description
A few navigation block tests were skipped, either due to being unstable or because they were temporarily skipped while the block underwent a lot of changes.

Now things are a bit more stable, it's a good time to bring these tests back.

### CODE REVIEW

File: packages/e2e-tests/specs/editor/blocks/__snapshots__/navigation.test.js.snap [hunk 1]
```diff
@@ -6,6 +6,8 @@ exports[`Navigation allows an empty navigation block to be created and manually
 <!-- wp:navigation-link {\\"label\\":\\"Contact\\",\\"type\\":\\"page\\",\\"id\\":[number],\\"url\\":\\"https://this/is/a/test/search/get-in-touch\\",\\"kind\\":\\"post-type\\",\\"isTopLevelLink\\":true} /-->"
 `;
 
+exports[`Navigation allows pages to be created from the navigation block and their links added to menu 1`] = `""`;
```
Comment (talldan) [inline packages/e2e-tests/specs/editor/blocks/__snapshots__/navigation.test.js.snap:9]: This is wrong, there should be a page list. It's working in #37358, so once that's merged I'll bring across those changes.

File: packages/e2e-tests/specs/editor/blocks/navigation.test.js [hunk 2]
```diff
@@ -564,125 +547,13 @@ describe( 'Navigation', () => {
 		expect( quickInserter ).toBeTruthy();
 	} );
 
-	// The following tests are unstable, roughly around when https://github.com/WordPress/wordpress-develop/pull/1412
-	// landed. The block manually tests well, so let's skip to unblock other PRs and immediately follow up. cc @vcanales
-	it.skip( 'loads frontend code only if the block is present', async () => {
```
Comment (talldan) [inline packages/e2e-tests/specs/editor/blocks/navigation.test.js:740]: I've divided this up into two tests.

File: packages/e2e-tests/specs/editor/blocks/navigation.test.js [hunk 3]
```diff
@@ -564,125 +547,13 @@ describe( 'Navigation', () => {
...
-					document.querySelectorAll(
-						'script[src*="navigation/view.min.js"]'
-					)
-				).length
-		);
-
-		expect( tagCount ).toBe( 1 );
-	} );
-
-	it.skip( 'loads frontend code only if responsiveness is turned on', async () => {
```
Comment (talldan) [inline packages/e2e-tests/specs/editor/blocks/navigation.test.js:800]: This is no longer true, so the test has been removed.

**Comment by talldan** [comment]
This should be ready for review now.

**Review by getdave — APPROVED** [review]
It's great to have all the tests back again. Thanks Dan!

**Pull request merged**